Update test, add server (#10)
ImmersiveLabsUkOrg/frontend-tech-test: README.md

# Immersive Labs Frontend Technical Test

This is the starter project for the Immersive Labs frontend technical test.

## Objective

- Use the provided API to create an app that retrieves and displays a list of users.
- For each user render a profile card. You need not be a design guru, but we do love attention to detail, so some layout, basic styling and user considerations will all be looked favourably upon.

## Requirements

### Basic

- Each card should display the name, avatar, email, joined date, bio and location of the user.
- The app should include a search field to search the list by the name of the user.
- The app should fetch the data from the server using either REST or GraphQL.
- You should write tests for the basic functionality of the app.

> 

Reviewer: ```suggestion
- We'd like to see a git commit history to understand the steps you took along the way
```

## Advanced

These are not part of the key requirements, but if they are too straightforward for you, or you just fancy showing off, here are some ideas for advanced features:

- If we perform the same search there may not be the need to query the API again. Implement a caching method to save the results of every search.
- To avoid calling the API every time a letter is entered into the search box add a 'debounce' function to buffer the input.

## Enhancements

> You are very welcome to enhance this functionality and make it as impressive and interactive as you like, but please do not exceed the time limit (2 hours, detailed below). Ideas for future enhancements can be discussed during the interview stage if not implemented during the test - we'd love to hear your ideas!

Reviewer: ```suggestion
You are very welcome to enhance this functionality and make it as impressive and interactive as you like, but please do not exceed the time limit (2 hours, detailed below). 

Ideas for future enhancements should be added to the readme.md and can be discussed during the interview stage - we'd love to hear your ideas!
```

## Tech

This project comes with Create React App, but you are free to use any boilerplate you wish or to start from scratch.

At Immersive Labs we use the following tech stack:

- [React.js](https://facebook.github.io/react/)
- [Styled Components](https://github.com/styled-components/styled-components)
- [Jest](https://facebook.github.io/jest/)
- [React Testing Library](https://testing-library.com)
- [TypeScript](https://www.typescriptlang.org/)
- [GraphQL](https://graphql.org/)
- [Apollo Client](https://www.apollographql.com/)

The project is pre-configured to allow use of these packages. You may use any of them you wish, or completely different technologies, but be prepared to discuss your decision.

## Time Limit

Please try and limit yourself to two hours on this test. We want to see what you can do in a short period of time and what you are comfortable with. We are not looking to see how many hours you are willing to put in.

If you run out of time but there are other things you would have liked to implement, make a note of these to discuss with us, or let us know via the `readme.md` of your project.

## Submission

> You can zip up your files and email them to [paul.thomas@immersivelabs.co.uk](mailto:paul.thomas@immersivelabs.co.uk) or submit via GitHub using a forked version of the repository and creating a pull request.

Reviewer: ```suggestion
Please zip up your files and return them to us by replying to the email you were sent.
```
